backend/app/src/user/user.gateway.ts

```diff
@@ -0,0 +1,57 @@
+import { UseGuards } from '@nestjs/common';
+import {
+  ConnectedSocket,
+  SubscribeMessage,
+  WebSocketGateway,
+  WebSocketServer,
+} from '@nestjs/websockets';
+import { UserStatus } from '@prisma/client';
+import { Server, Socket } from 'socket.io';
+import { JwtAuthGuard } from '../auth/guard/jwt.auth-guard';
+import { GetCurrentUserId } from '../common/decorators/getCurrentUserId.decorator';
+import { UserService } from './user.service';
+
+@WebSocketGateway({
+  cors: {
+    origin: 'http://localhost:8080',
+    credentials: true,
+  },
+})
+@UseGuards(JwtAuthGuard)
+export class UserGateway {
+  @WebSocketServer()
+  server: Server;
+  constructor(private readonly userService: UserService) {}
+
+  @SubscribeMessage('connectUser')
+  async userConnect(
+    @GetCurrentUserId() userId: string,
+    @ConnectedSocket() clientSocket: Socket,
+  ) {
+    await this.userService.updateConnectionStatus(userId, UserStatus.ONLINE);
```

Comment:
This return in not error handled, and will emit the same thing wether or not the db write was succesful or not, is that the desired effect?